instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -146,15 +152,27 @@ def process_request(self, request):
 
         request_meta = request.META
 
-        token = attach(extract(request_meta, getter=wsgi_getter))
+        context = extract(request_meta, getter=wsgi_getter)
+
+        token = attach(context)
```

Comment:
I'm a bit confused by this. Aren't we still activating the extracted span here and wouldn't `get_current_span()` call below always return that activated span? 

What we need to do this to call `get_current_span()` first and if it is invalid, then try to extract a span from the carrier and activate it. After that we just call tracer.start_span either with INTERNAL or SERVER kind. We'd probably also need to update the detach calls to do it conditionally.